#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/Parquetry.proto"

WorldInfo {
  defaultDamping Damping {
  }
}
Viewpoint {
  orientation 0.17152773171417762 0.17152773171417768 -0.9701321943456872 1.6011146149123774
  position 8.420553004483057e-15 7.7502370472865545 2.8729949022303467
  follow "icarus"
  followType "Pan and Tilt Shot"
}
TexturedBackground {
  texture "noon_park_empty"
}
TexturedBackgroundLight {
}
DEF props Group {
  children [
    RectangleArena {
      rotation 0 1 0 0
      floorSize 5 5
      floorAppearance Parquetry {
        type "light strip"
      }
      wallThickness 0.1
    }
  ]
}
DEF _ Robot {
  translation 0 0 0.0495096
  children [
    Accelerometer {
    }
    InertialUnit {
      name "imu"
    }
    GPS {
    }
    DEF propRB Propeller {
      shaftAxis 0 0 1
      centerOfThrust -0.24 -0.24 0.02
      thrustConstants 1.96503e-05 0
      torqueConstants 3.22606e-07 0
      device RotationalMotor {
        name "motorRB"
        maxVelocity 6000
        maxTorque 1000
      }
      fastHelix DEF prop_fast Solid {
        translation -0.24 -0.24 0.02
        rotation 0 0 1 1.1959338306901244
        children [
          DEF prop1155_fast Shape {
            appearance PBRAppearance {
              transparency 0.7
              roughness 0.9
              metalness 0.1
            }
            geometry Cylinder {
              height 0.005
              radius 0.14
            }
          }
        ]
      }
      slowHelix DEF prop_slow Solid {
        translation -0.24 -0.24 0.02
        rotation 0 0 -1 4.367852913909413
        children [
          DEF prop1155 Shape {
            appearance PBRAppearance {
              baseColor 0.180392 0.203922 0.211765
              roughness 0.5
              metalness 0.1
            }
            geometry Box {
              size 0.01 0.279 0.003
            }
          }
        ]
      }
    }
    DEF propLB Propeller {
      shaftAxis 0 0 -1
      centerOfThrust -0.24 0.24 0.02
      thrustConstants -1.96503e-05 0
      torqueConstants 3.22606e-07 0
      device RotationalMotor {
        name "motorLB"
        maxVelocity 6000
        maxTorque 1000
      }
      fastHelix DEF prop_fast Solid {
        translation -0.24 0.24 0.02
        rotation 0 0 -1 1.3685486227416819
        children [
          USE prop1155_fast
        ]
      }
      slowHelix DEF prop_slow Solid {
        translation -0.24 0.24 0.02
        rotation 0 0 1 4.403316458918791
        children [
          USE prop1155
        ]
      }
    }
    DEF propRF Propeller {
      shaftAxis 0 0 -1
      centerOfThrust 0.24 -0.24 0.02
      thrustConstants -1.96503e-05 0
      torqueConstants 3.22606e-07 0
      device RotationalMotor {
        name "motorRF"
        maxVelocity 6000
        maxTorque 1000
      }
      fastHelix DEF prop_fast Solid {
        translation 0.24 -0.24 0.02
        rotation 0 0 -1 1.1239513311678
        children [
          USE prop1155_fast
        ]
      }
      slowHelix DEF prop_slow Solid {
        translation 0.24 -0.24 0.02
        rotation 0 0 1 4.648606445464477
        children [
          USE prop1155
        ]
      }
    }
    DEF propLF Propeller {
      shaftAxis 0 0 1
      centerOfThrust 0.24 0.24 0.02
      thrustConstants 1.96503e-05 0
      torqueConstants 3.22606e-07 0
      device RotationalMotor {
        name "motorLF"
        maxVelocity 6000
        maxTorque 1000
      }
      fastHelix DEF prop_fast Solid {
        translation 0.24 0.24 0.02
        rotation 0 0 -1 3.841550218014254
        children [
          USE prop1155_fast
        ]
      }
      slowHelix DEF prop_slow Solid {
        translation 0.24 0.24 0.02
        rotation 0 0 1 2.9522289307774248
        children [
          USE prop1155
        ]
      }
    }
    DEF base Pose {
      children [
        Shape {
          appearance DEF carbonFiber PBRAppearance {
            baseColor 0.184314 0.184314 0.184314
            roughness 0.5
            metalness 0.8
          }
          geometry Cylinder {
            height 0.1
            radius 0.125
          }
        }
      ]
    }
    DEF rodLeft Pose {
      rotation 0.7071067811865475 0.7071067811865475 0 1.57079
      children [
        DEF rod Shape {
          appearance USE carbonFiber
          geometry Cylinder {
            height 0.7
            radius 0.015
          }
        }
      ]
    }
    DEF rodRight Pose {
      rotation -0.7071067811865475 0.7071067811865475 0 1.57079
      children [
        DEF rod Shape {
          appearance USE carbonFiber
          geometry Cylinder {
            height 0.7
            radius 0.015
          }
        }
      ]
    }
  ]
  name "icarus"
  boundingObject Group {
    children [
      USE rodRight
      USE rodLeft
      USE base
    ]
  }
  physics Physics {
    density -1
    mass 2.5
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      0.102757 0.102757 0.185948
      0 0 0
    ]
  }
  controller "robot_supervisor_controller"
  supervisor TRUE
}
